docs/gene_sets.md

```diff
@@ -0,0 +1,110 @@
+## Overview
+
+A gene set is a named list of user-ordered genes. A gene set can describe marker genes for a cell type, state, or pathway.
```

Comment:
I'm not sure if `user-ordered genes` is the clearest concept...it may be clearer to say "ordered list of genes".  

Suggest something like:   

> A gene set is a user-created, ordered list of genes identifiers.  Typical uses for a gene set include marker genes for a cell type, state, or pathway.  
> 
> A gene set must have a unique name.  A given gene may be included in multiple gene sets. A gene set may optionally include a description for the entire gene set, and descriptions for each gene included the gene set.